src/Records/Types/Long.php

```diff
@@ -38,10 +38,12 @@ public function setValue($value)
     {
         $value = (int)$value;
 
-        if ($value < 0) {
-            throw new \UnderflowException('Long integer value must be in the range 0 - 4294967296');
-        } else if ($value > 0xffffffff) {
-            throw new \OverflowException('Long integer value must be in the range 0 - 4294967296');
+        if (4 !== \PHP_INT_SIZE) {
+            if ($value < 0) {
+                throw new \UnderflowException('Long integer value must be in the range 0 - 4294967296');
+            } else if ($value > 0xffffffff) {
+                throw new \OverflowException('Long integer value must be in the range 0 - 4294967296');
+            }
         }
```

Comment:
If we're on 32 bit, values that should be positive (unsigned) will be negative, no?

Reply:
On 32-bit, we won't enter this branch because of the `4 !== \PHP_INT_SIZE` check above.

Reply:
I know, but values decoding to large values will be negative on 32 bit, no? Shouldn't we reject those?

Reply:
Nevermind, I guess it's fine that way.